Source/CombatExtended/CombatExtended/CE_Utility.cs

```diff
@@ -997,7 +997,7 @@ public static bool IntersectionPoint(Vector3 p1, Vector3 p2, Vector3 center, flo
             Vector3 lp2 = p2 - center;
 
             // If we obviously don't cross, early out.
-            if (lp1.sqrMagnitude < radSq && lp2.sqrMagnitude < radSq)
+            if (lp1.sqrMagnitude > radSq && lp2.sqrMagnitude > radSq)
```

Comment:
You can't early out here, because you could have two intersection points.  If you have two intersection points, then the bullet both starts (lp1.sqrMagnitude) and stops (lp2.sqrMagnitude) outside the radius, but passes through in between.  

Instead, the case where both those are greater, you must look at the line segment's closest point and see if *it* is inside.  Or you can look at the relative direction vector at position 1 and position 2 and see if it is headed toward the local origin in position 1, and away in position 2.